packages/block-editor/src/hooks/color.js

```diff
@@ -219,13 +219,16 @@ export function ColorEdit( props ) {
 	const {
 		palette: solidsPerOrigin,
 		gradients: gradientsPerOrigin,
-		customGradient: areCustomGradientsEnabled,
-		custom: areCustomSolidsEnabled,
 		text: isTextEnabled,
 		background: isBackgroundEnabled,
 		link: isLinkEnabled,
 	} = useSetting( 'color' ) || {};
 
+	// The following color settings need to be accessed explicitly due to
```

Comment:
Oh, that's a good one. It sounds like a follow-up should be to cover for this case in `useSetting` (so the deprecated flags are still considered when accessing the top-level keys `color`, `typography`, `spacing`).

Reply:
https://github.com/WordPress/gutenberg/issues/37094

FYI @jorgefilipecosta @youknowriad

Reply:
While it can be a bit more rare given that themes provide the settings via `theme.json` or we retrofit them, would be worth doing the same for `color.palette` and `color.gradients` (accessing them directly here) just to be sure?

Reply:
Should we deprecate useSetting with top level keys? `color`, `typography`, `spacing`. I feel this was never considered to be a valid usage.

If we want to keep them, we'd need to "normalize" the settings before running the useSetting logic.

Reply:
I've reverted accessing `color` directly for all the changes I've found https://github.com/WordPress/gutenberg/pull/36900/commits/6a825e63e23bc65ae98983a5f2061a62e6ff00e6

I know this wasn't directly related to this bug but it's the source of the bug, so it's best addressed together. It's also one less PR to backport.

Reply:
> (Andrew, I hope you don't mind I have pushed directly to this PR. I did it so timezones work in our favor, not against).

@oandregal not at all! Thanks so much for picking up this PR, it's always good to get timezones to work in our favor, much appreciated 🙇